tests/test_assembleCoadd.py

```diff
@@ -0,0 +1,261 @@
+                                  "meant to test debugging options.")
+
+    def runQuantum(self, mockSkyInfo, mockRefList, *args):
+        """Simplified API for testing image coaddition algorithms.
+
+        Parameters
+        ----------
+        mockSkyInfo : `lsst.pipe.base.Struct`
+            A simple container that supplies a bounding box and WCS in the
+            same format as the output of
+            `lsst.pipe.tasks.CoaddBaseTask.getSkyInfo`
+        mockRefList : `list` of `lsst.pipe.tasks.MockGen2ExposureReference`
+            Description
+
+        Returns
+        -------
+        retStruct : `lsst.pipe.base.Struct`
+            The coadded exposure and associated metadata.
+        """
+        inputs = self.prepareInputs(mockRefList)
+        supplementaryData = self.makeSupplementaryData(mockSkyInfo, warpRefList=inputs.tempExpRefList)
+
+        retStruct = self.run(mockSkyInfo, inputs.tempExpRefList, inputs.imageScalerList,
+                             inputs.weightList, supplementaryData=supplementaryData)
+        return retStruct
+
+    def runDataRef(self, mockSkyInfo, selectDataList=None, warpRefList=None):
+        """Simplified API for testing image coaddition algorithms.
+
+        Notes
+        -----
+        This tests the Gen 2 Butler interface, and can be removed once that
+        is fully deprecated.
+
+        Parameters
+        ----------
+        mockSkyInfo : `lsst.pipe.base.Struct`
+            A simple container that supplies a bounding box and WCS in the
+            same format as the output of
+            `lsst.pipe.tasks.CoaddBaseTask.getSkyInfo`
+        warpRefList : `list` of `lsst.pipe.tasks.MockExposureReference`
+            Description
+
+        Returns
+        -------
+        retStruct : `lsst.pipe.base.Struct`
+            The coadded exposure and associated metadata.
+        """
+        inputData = self.prepareInputs(warpRefList)
+        supplementaryData = self.makeSupplementaryData(mockSkyInfo, warpRefList=inputData.tempExpRefList)
+        retStruct = self.run(mockSkyInfo, inputData.tempExpRefList, inputData.imageScalerList,
+                             inputData.weightList, supplementaryData=supplementaryData)
+        return retStruct
+
+
+class MockSafeClipAssembleCoaddConfig(SafeClipAssembleCoaddConfig):
+
+    def setDefaults(self):
+        SafeClipAssembleCoaddConfig.setDefaults(self)
+        self.assembleMeanCoadd.retarget(MockAssembleCoaddTask)
+        self.assembleMeanClipCoadd.retarget(MockAssembleCoaddTask)
+        self.doWrite = False
+
+
+class MockSafeClipAssembleCoaddTask(MockAssembleCoaddTask, SafeClipAssembleCoaddTask):
+
+    """Lightly modified version of `SafeClipAssembleCoaddTask`
+    for use with unit tests.
+    """
+    ConfigClass = MockSafeClipAssembleCoaddConfig
+    _DefaultName = "safeClipAssembleCoadd"
+
+    def __init__(self, *args, **kwargs):
+        SafeClipAssembleCoaddTask.__init__(self, *args, **kwargs)
```

Comment:
This one's trickier because of multiple inheritance. Is there a way to get `super` to pick up `__init__` from `SafeClipAssembleCoaddTask ` instead of `MockAssembleCoaddTask`?

Reply:
Do you not want it to call `MockAssembleCoaddTask.__init__()` at all, or just after the other one? The Method Resolution Order (MRO) is determined by the order of the parent classes in the class definition, so `MockAssembleCoaddTask, SafeClipAssembleCoaddTask` calls parent methods in that order with `super`, and `SafeClipAssembleCoaddTask, MockAssembleCoaddTask` calls them in the other order.

If you specifically only want the one and not the other, you can make a specific call like this, but please note it with a comment as to why you didn't use `super`, so someone doesn't replace it in the future.